iron-input.html

```diff
@@ -125,11 +175,50 @@
      * @suppress {checkTypes}
      */
     _bindValueChanged: function() {
+      if (typeof this.bindValue !== 'string') {
+        // it will trigger another _bindValueChanged event
+        this.bindValue = !(this.bindValue || this.bindValue === 0) ? '' : this.bindValue + '';
+      } else {
+        this._updateValueIfNeeded();
+        // manually notify because we don't want to notify until after setting value
+        this.fire('bind-value-changed', {value: this.bindValue});
+      }
+    },
+
+    _bindValueAsNumberChanged: function() {
+      // ensure it is a number
+      if (typeof this.bindValueAsNumber !== 'number') {
+        // it will trigger another _bindValueAsNumberChanged event
+        // parseFloat would convert '1a' to 1, so we use window.isNaN to handle these cases
+        this.bindValueAsNumber = (typeof this.bindValueAsNumber === 'string' && !window.isNaN(this.bindValueAsNumber)) ? parseFloat(this.bindValueAsNumber) : NaN;
```

Comment:
@cdata in this case we have to use <code>window.isNaN</code> because:
```javascript
window.isNaN('1a'); // returns true
Number.isNaN('1a'); // returns false